src/util/fpscounter.rs

```diff
@@ -0,0 +1,62 @@
+use ::time;
+use ::datastructure;
+use std::collections::VecDeque;
+use amethyst::ecs::System;
+
+///The FPSCounter resource needed by the FPSCounterSystem
+///Add it to your resources to be able to use the FPSCounterSystem
+///world.add_resource(FPSCounter::new(20));
+///Usage example:
+///Get the FPSCounter resource.
+///println!("Cur FPS: {}, Sampled: {}",counter.frame_fps(),counter.sampled_fps());
+pub struct FPSCounter{
+    buf:CircularBuffer<f32>,
+}
+impl FPSCounter{
+    ///Creates a new FPSCounter that calculates the average fps over samplesize values.
+    pub fn new(samplesize:usize)->FPSCounter{
+        FPSCounter{
+            buf: CircularBuffer::<f32>::new(samplesize),
+        }
+    }
+    ///Add a new delta time value.
+    pub fn push(&mut self,elem:f32){
+        self.buf.push(elem);
+    }
+    ///Get the fps of the this frame.
+    pub fn frame_fps(&self)->f32{
+        if let Some(back) = self.buf.queue().back(){
+            return 1.0 / *back
+        }
+        0.
+    }
+    ///Get the average fps over the samplesize frames.
+    pub fn sampled_fps(&self)->f32{
+        let mut total = 0.0 as f32;
+        for e in self.buf.queue().iter(){
+            total += *e;
+        }
+        1.0 / (total / self.buf.queue().len() as f32)
+    }
+}
+
+///FPSCounterSystem
+///Add this system to your game to calculate FPS
+pub struct FPSCounterSystem;
```

Comment:
Can we keep ECS code under the ecs package ?
Put this under ecs::util or ecs::fps or something like that.

Reply:
Fixed (I think)